### DISCUSSION TITLE
Custom plugin not being discovered/loaded.

### DISCUSSION THREAD

**Discussion by j-huff** [inventree/InvenTree#7233] [Q&A] [answered]
I am almost certain I'm doing everything correctly and I'm beginning to go insane.

I'm trying to make an extremely simple plugin that uses the report mixin. I have the file plugins/stock_location_scaling/stock_label_width.py with the following content:

```
from plugin import InvenTreePlugin
from plugin.mixins import ReportMixin
from django.utils.translation import gettext_lazy as _

class StockLocationScaling(ReportMixin, InvenTreePlugin):
    NAME = "StockLocationScaling"
    SLUG = "stocklocationscaling"
    TITLE = _("Stock Location Scaling")
    DESCRIPTION = _("A plugin which provides extra context data to a report")
    VERSION = "1.0"
    AUTHOR = "John Huff"

    def add_label_context(self, label_instance, model_instance, request, context):
        context["test_test"] = "THIS IS SOME TEXT"

    def add_report_context(self, report_instance, model_instance, request, context):
        """Add example content to the report instance"""
        context["test_test2"] = "THIS IS SOME TEXT"
```

I also have an empty `__init__.py` file in that directory.

I'm using the inventree docker production server and have set INVENTREE_PLUGINS_ENABLED=True in the .env, and I've set plugins_enabled=True in the ext/config.yaml as well (just in case i guess...). I've restarted the server using `docker compose restart`.

When I go print a label where the template contains the following:

```
<span class='text'>Text: {{ test_test2 }} </span>
```

All I see is "Text:". test_test2 doesn't seem to be added to the context, nor test_test. 

What am I missing? Why can't I load a simple custom plugin? I'm lost.

The plugin also doesn't show up in the  Plugin Configurations admin page.

**Comment by j-huff** [comment]
I can get the plugin to work/show up if I throw it in `plugin/builtin`, but that's certainly not ideal...

**Comment by matmair** [comment, marked as answer]
Sounds like the plugin is not discovered. If it is not showing up in the settings/plugin table InvenTree does not know about it.
Could be a number of things but the most probable is that module discovery itself is not working.
1) Do you have an empty `__init__.py` in `plugins` and `stock_location_scaling`?
2) If yes: place `raise KeyError('test)` in both files, restart the server and check if you see discovery errors in the settings/plugin pane

**Reply by j-huff** [reply]
It's working now (being discovered), but I could not for the life of me tell you why. Thanks for the debugging suggestion anyway.
